components/camel-wmq/src/main/java/org/apacheextras/camel/component/wmq/WMQComponent.java

```diff
@@ -118,4 +120,13 @@ public MQQueueManager getQueueManager(String name, String hostname, String port,
         }
     }
 
+    @Override
+    public void close() throws IOException {
+        try {
+            WorkQueueManager.close();
+        } catch (CSIException e) {
+            throw new RuntimeException(e);
+        }
+        super.close();
```

Comment:
can we wrap the super.close() inside a finally block ? to be sure we close parent resource even if we have a runtime exception.

Reply:
of course - good point; updated